src/mem.rs

```diff
@@ -0,0 +1,60 @@
+// NOTE Copied verbatim from the rlibc crate
```

Comment:
Could you elaborate on what the bug is? Is it present on all architectures or only on AVR where `usize` is 16-bits(?).